tests/Microsoft.Identity.Test.Unit/TelemetryTests/HttpTelemetryTests.cs

```diff
@@ -390,6 +361,87 @@ await pca.AcquireTokenByUsernamePassword(TestConstants.s_scope, "username", Test
             }
         }
 
+        [TestMethod]
+        public async Task CallerSdkDetailsTestAsync()
+        {
+            using (_harness = CreateTestHarness())
+            {
+                _harness.HttpManager.AddInstanceDiscoveryMockHandler();
+                var requestHandler = _harness.HttpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+                var cca = CreateConfidentialClientApp();
+
+                await cca.AcquireTokenForClient(TestConstants.s_scope)
+                    .WithExtraQueryParameters(new Dictionary<string, string> { 
+                        { "caller-sdk-id", "testApiId" },
+                        { "caller-sdk-ver", "testSdkVersion"} })
+                    .ExecuteAsync().ConfigureAwait(false);
+
+                AssertCurrentTelemetry(
+                    requestHandler.ActualRequestMessage, 
+                    ApiIds.AcquireTokenForClient, 
+                    CacheRefreshReason.NoCachedAccessToken,
+                    callerSdkId: "testApiId", 
+                    callerSdkVersion: "testSdkVersion");
+            }
+        }
+
+        [TestMethod]
+        public async Task CallerSdkDetails_ConstraintsTestAsync()

```

Comment:
do we have a ` CallerSdkDetails_NullValuesTestAsync` test to check for null id and versionb?